addons/sale_timesheet/report/project_profitability_report_analysis.py

```diff
@@ -88,39 +72,52 @@ def init(self):
                             SUM(timesheet_cost) AS timesheet_cost,
                             SUM(expense_cost) AS expense_cost,
                             SUM(other_revenues) AS other_revenues,
-                            SUM(downpayment_invoiced) AS downpayment_invoiced
+                            SUM(downpayment_invoiced) AS downpayment_invoiced,
+                            SUM(expense_amount_untaxed_to_invoice) AS expense_amount_untaxed_to_invoice,
+                            SUM(expense_amount_untaxed_invoiced) AS expense_amount_untaxed_invoiced,
+                            SUM(amount_untaxed_to_invoice) AS amount_untaxed_to_invoice,
+                            SUM(amount_untaxed_invoiced) AS amount_untaxed_invoiced,
+                            line_date AS line_date
                         FROM (
                             SELECT
                                 P.id AS project_id,
                                 P.analytic_account_id AS analytic_account_id,
                                 TS.so_line AS sale_line_id,
-                                SUM(TS.unit_amount) AS timesheet_unit_amount,
-                                SUM(TS.amount) AS timesheet_cost,
+                                TS.unit_amount AS timesheet_unit_amount,
+                                TS.amount AS timesheet_cost,
                                 0.0 AS other_revenues,
                                 0.0 AS expense_cost,
-                                0.0 AS downpayment_invoiced
+                                0.0 AS downpayment_invoiced,
+                                0.0 AS expense_amount_untaxed_to_invoice,
+                                0.0 AS expense_amount_untaxed_invoiced,
+                                0.0 AS amount_untaxed_to_invoice,
+                                0.0 AS amount_untaxed_invoiced,
+                                TS.date AS line_date
                             FROM account_analytic_line TS, project_project P
                             WHERE TS.project_id IS NOT NULL AND P.id = TS.project_id AND P.active = 't' AND P.allow_timesheets = 't'
-                            GROUP BY P.id, TS.so_line
 
-                            UNION
+                            UNION ALL
```

Comment:
Why UNION ALL. 

You need the duplicates ?

Reply:
I need each record of all `SELECT` to be independent.
With just `UNION` some records were merged and were generating incorrect values which didn't pass the tests.

Reply:
Ok :)

And are we sure that we don't count the same line twice then ? Just to make sure.

Reply:
It is possible that the same line is split into the different values, generating partially duplicated lines.
For example, the same line could be split to 2 lines:
- one having a value for `timesheet cost` and the other values set to 0.0
- one having a value for `expense cost` and the other values set to 0.0
Unfortunately if `timesheet cost` and `expense cost` are both equal to 0.0, then there will be 2 totally duplicated lines.